[Defend Workflows][Osquery] Multiple fixes for pack management

szwarckonrad opened the pull request:
This PR combines three separate osquery pack management fixes:

1. **Fix global packs not preserving policy references**: Resolves issues where global packs would lose references to existing policies when creating new osquery integrations or deleting policies. Fixed by including `pack.references` in SavedObject mapping and correcting the filter logic in the delete callback.


https://github.com/user-attachments/assets/452b52b3-198a-428f-a63e-e33857bfa930

2. **Fix pack edit cancel button**: Fixes the cancel button on the pack edit page that was redirecting to `/packs/undefined` instead of the correct pack details page.

3. **Update pack confirmation modal**: Improves the confirmation modal to clearly display both the number of agent policies and the number of agents that will be affected by pack updates.

<img width="581" height="300" alt="Screenshot 2025-10-31 at 16 47 11" src="https://github.com/user-attachments/assets/fa14067f-de8f-4377-8898-411a0454fe91" />

Closes https://github.com/elastic/security-team/issues/14422
Closes https://github.com/elastic/security-team/issues/14423
Closes https://github.com/elastic/security-team/issues/14424


<!--ONMERGE {"backportTargets":["8.19","9.1","9.2"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
Pinging @elastic/security-defend-workflows (Team:Defend Workflows)

szwarckonrad committed:
[Defend Workflows][Osquery] Multiple fixes for pack management

- Fix global packs not preserving policy references when creating new integrations or deleting policies
- Fix pack edit cancel button redirecting to undefined page
- Update pack confirmation modal to display both policy and agent counts
- Update live query route privileges to use readLiveQueries

Closes https://github.com/elastic/security-team/issues/14422
Closes https://github.com/elastic/security-team/issues/14423
Closes https://github.com/elastic/security-team/issues/14424

szwarckonrad committed:
Merge branch 'main' into osquery-small-fixes

paul-tavares reviewed (approved):
👍

kibanamachine commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.19|Backport failed because of merge conflicts|
|❌|9.1|**Backport failed because of merge conflicts**<br><br>You might need to backport the following PRs to 9.1:<br> - [[ResponseOps][Maintenance Window] Timezone field disappears when specific timezone is selected (#241574)](https://github.com/elastic/kibana/pull/241574)<br> - [Fix enforcement of unique policy name of policies across spaces (#239631)](https://github.com/elastic/kibana/pull/239631)<br> - [[SLO] exclude stale SLOs from group by stats (#240077)](https://github.com/elastic/kibana/pull/240077)<br> - [[Observability Onboarding] Copy update for landing page messages (#241767)](https://github.com/elastic/kibana/pull/241767)|
|✅|9.2|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/241959">](https://github.com/elastic/kibana/pull/241959)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 241655
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

szwarckonrad commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/241962">](https://github.com/elastic/kibana/pull/241962)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.1.0 -->
